File: mns/python/entrance.py
# -*- coding: utf-8 -*-
import json
import logging

from aliyunsdkcore.client import AcsClient
from aliyunsdkcore.auth.credentials import StsTokenCredential
from aliyunsdkfnf.request.v20190315.StartExecutionRequest import StartExecutionRequest


def handler(event, context):
    logger = logging.getLogger()

    # Parse string format event to json
    evt = json.loads(event)
    logger.info('Receive mns trigger event: %s', evt)

    # Read sts ak/sk/token from context and construct a AcsClient
    creds = context.credentials
    sts_creds = StsTokenCredential(creds.access_key_id, creds.access_key_secret, creds.security_token)
    client = AcsClient(credential=sts_creds, region_id=context.region)

    # StartExecution api to execute serverless workflow
    flow_name = 'fnf-demo-mns-flow'
    req = StartExecutionRequest()
    req.set_FlowName(flow_name)
    req.set_Input(event)

    resp = client.do_action_with_exception(req)
    logger.info('StartExecution flow: %s, response: %s', flow_name, resp)

    return "success"

File: mns/php/entrance.php
<?php

use AlibabaCloud\Client\AlibabaCloud;
use AlibabaCloud\Fnf\Fnf;
use AlibabaCloud\Client\Credentials\StsCredential;
use AlibabaCloud\Client\Signature\ShaHmac1Signature;


function handler($event, $context) {
    $eventObj = json_decode($event, $assoc = true);
    $logger = $GLOBALS['fcLogger'];
    $logger->info('Receive mns trigger event: '.$event);

    # Read ak/sk/token from context and construct a invoke client
    $stsCreds = new StsCredential($context['credentials']['accessKeyId'],
        $context['credentials']['accessKeySecret'],
        $context['credentials']['securityToken']);
    AlibabaCloud::client($stsCreds, new ShaHmac1Signature())
        ->regionId($context['region'])
        ->asDefaultClient();

    # StartExecution to execute serverless workflow
    $flowName = 'fnf-demo-mns-flow';
    $options = [
        'debug'=>false,
        'connect_timeout'=>10,
        'timeout'=>10,
        'form_params'=>[
            'FlowName'=>$flowName,
            'Input'=>json_encode($eventObj),
        ],
    ];
    $resp = Fnf::v20190315()
        ->startExecution($options)
        ->request();
    $logger->info('StartExecution flow:'.$flowName.', response: '.(string)$resp);

    return 'hello world';
}